### ISSUE TITLE
"Unexpected Token" for async arrow IIFE

### ISSUE THREAD

**Issue by spalger** [babel/babel-eslint#245]
Using eslint `4.1.7` I am getting a syntax error with this code:

```js
(async () => {

}());
```

> Unexpected token at line 3 col 2

**Comment by spalger** [comment]
This only occurs on arrow functions. `(async function () {}());` works fine.

**Comment by jmm** [comment]
I don't think that input is valid. `AsyncArrowFunction` is an `AssignmentExpression` like `ArrowFunction` and this is a syntax error:

```js
() => {}();
```

It has to be 

```js
(() => {})();
```

There was a bug in Babel 5.x that allowed the former, but apparently it doesn't extend to async arrows.

**Comment by spalger** [comment]
hmm, interesting
